File: src/elixir/ebpf_translator/lib/examples/minimal.c
#include <linux/bpf.h>

#include <bpf/bpf_helpers.h>

char LICENSE[] SEC("license") = "Dual BSD/GPL";

SEC("tp/syscalls/sys_enter_write")
int main() {

  int hmm0nil = "?";

  int pid1nil = bpf_get_current_pid_tgid();

  bpf_printk("BPF triggered from PID %d.\n", pid1nil);

  return 0;
}

File: src/elixir/ebpf_translator/lib/examples/minimal.ex
defmodule T do
  use Ebpf, [license: "Dual BSD/GPL"]

  import Linux.Bpf
  import Bpf.Bpf_helpers

  @sec "tp/syscalls/sys_enter_write"
  def main do
    hmm = "?"
    pid = bpf_get_current_pid_tgid()
    bpf_printk(["BPF triggered from PID %d.\n", pid])
    0
  end
end
